core/trino-main/src/test/java/io/trino/operator/scalar/TestUrlFunctions.java

```diff
@@ -168,4 +179,33 @@ private void validateUrlExtract(String url, String protocol, String host, Long p
                 .hasType(createVarcharType(url.length()))
                 .isEqualTo(fragment);
     }
+
+    private void invalidUrlExtract(String url)
+    {
+        checkArgument(!url.contains("'")); // Would require escaping in literals
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_protocol", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_host", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_port", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_path", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_query", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
+
+        assertTrinoExceptionThrownBy(() -> assertions.function("url_extract_fragment", "'" + url + "'").evaluate())
+                .hasErrorCode(INVALID_FUNCTION_ARGUMENT)
+                .hasMessageStartingWith("Cannot parse as URI value '%s': ".formatted(url));
```

Comment:
test also `try(url_extract_fragment(...`

Reply:
i don't think we should be testing `try` in `TestUrlFunctions.java`